data-processing-lib/spark/src/data_processing_spark/runtime/spark/spark_launcher.py

```diff
@@ -0,0 +1,219 @@
+    DataAccessFactory,
+    DataAccessFactoryBase,
+)
+from data_processing.runtime import (
+    AbstractTransformLauncher,
+    TransformExecutionConfiguration,
+)
+from data_processing.utils import get_logger, str2bool
+from data_processing_spark.runtime.spark.runtime_config import (
+    SparkTransformRuntimeConfiguration,
+)
+from data_processing_spark.runtime.spark.spark_execution_config import (
+    SparkExecutionConfiguration,
+)
+from data_processing_spark.runtime.spark.spark_transform import AbstractSparkTransform
+from pyspark.sql import DataFrame, SparkSession
+
+
+logger = get_logger(__name__)
+
+
+class SparkTransformLauncher(AbstractTransformLauncher):
+    """
+    Driver class starting Filter execution
+    """
+
+    def __init__(
+        self,
+        runtime_config: SparkTransformRuntimeConfiguration,
+        data_access_factory: DataAccessFactoryBase = DataAccessFactory(),
+    ):
+        """
+        Creates driver
+        :param runtime_config: transform runtime factory
+        :param data_access_factory: the factory to create DataAccess instances.
+        """
+        super().__init__(runtime_config, data_access_factory)
+        self.runtime_config = runtime_config
+        self.execution_config = SparkExecutionConfiguration(runtime_config.get_name())
+
+    def launch(self):
+        if not self._get_args():
+            logger.warning("Arguments could not be applied.")
+            return 1
+        transform_params = dict(self.runtime_config.get_transform_params())
+        transform_class = self.runtime_config.get_transform_class()
+        transform = transform_class(transform_params)
+        data_access = self.data_access_factory.create_data_access()
+        self._start_spark()
+        self._run_transform(data_access, transform)
+        self._stop_spark()
+
+    def _start_spark(self):
+        server_port_https = int(os.getenv("KUBERNETES_SERVICE_PORT_HTTPS", "-1"))
+        if server_port_https == -1:
+            # we are running locally, use the spark_profile_local.yaml file for configuration
+            config_filepath = self.execution_config.local_config_filepath
+            # config_filepath = os.path.abspath(
+            #     os.path.join(
+            #         os.path.dirname(__file__), "../../../../", "test-data", "config", "spark_profile_local.yaml"
+            #     )
+            # )
+            with open(config_filepath, "r") as config_fp:
+                spark_config = yaml.safe_load(os.path.expandvars(config_fp.read()))
+            app_name = spark_config.get("spark.app.name", "my-spark-app")
+            self.spark = SparkSession.builder.appName(app_name).config(map=spark_config).getOrCreate()
+        else:
+            # we are running in Kubernetes, use spark_profile.yaml and
+            # environment variables for configuration
+
+            server_port = os.environ["KUBERNETES_SERVICE_PORT"]
+            master_url = f"k8s://https://kubernetes.default:{server_port}"
+
```

Comment:
See comment above. I do not think it is the right approach. We should bite the bullet and adopt external configuration https://github.com/kubeflow/spark-operator/blob/master/docs/user-guide.md. This will make code smaller and provide a better configurability moving forward

Reply:
Your point here is a valid one and we should consider it after we get this initial implementation in.  Is there any reason why we should not have this initial implementation, even if we decide to deprecate it in the future?  Would it somehow make all the class/launcher design obsolete? If yes, can you say why?